lib/pbench/test/unit/server/query_apis/test_datasets.py

```diff
@@ -0,0 +1,612 @@
+            "throttled_until_millis": 0,
+            "failures": ["bad", "me too"],
+        }
+        query_api(
+            "/datasets/random_md5_string1?access=public",
+            "/_update_by_query?ignore_unavailable=true&refresh=true",
+            payload=None,
+            expected_index=index,
+            expected_status=HTTPStatus.INTERNAL_SERVER_ERROR,
+            request_method=ApiMethod.POST,
+            json=es_json,
+            headers=headers,
+            expect_call=True,
+        )
+
+    @pytest.mark.parametrize("ao", (True, False))
+    def test_get(
+        self, monkeypatch, more_datasets, client, query_api, build_auth_header, ao
+    ):
+        """Check on the GET summary behavior
+
+        We should report a JSON document with an integer count for each index
+        name.
+
+        We test two different datasets, one with an index map and one without,
+        both with and without "archiveonly" set.
+        """
+        auth_json = {"user": "drb", "access": "private"}
+
+        expected_status = self.get_expected_status(
+            auth_json, build_auth_header["header_param"]
+        )
+
+        ds = Dataset.query(name="drb")
+        json = copy.deepcopy(self.empty_es_response_payload)
+        if ao:
+            # Set archiveonly flag to disable index-map logic
+            Metadata.setvalue(ds, Metadata.SERVER_ARCHIVE, True)
+            index = None
+            if expected_status == HTTPStatus.OK:
+                expected_status = HTTPStatus.CONFLICT
+            monkeypatch.setattr(IndexMap, "indices", lambda _d: [])
+            expected = {}
+        else:
+            indices = list(IndexMap.indices(ds))
+            index = "/" + ",".join(indices)
+            hits = []
+            expected = {}
+            for i, n in enumerate(indices):
+                hits.append({"_index": n, "_id": i, "_source": {"data": f"{n}_{i}"}})
+                expected[n] = 1
+            json["hits"]["total"]["value"] = str(len(hits))
+            json["hits"]["hits"] = hits
+
+        response = query_api(
+            self.pbench_endpoint,
+            "/_search?ignore_unavailable=true",
+            self.payload,
+            index,
+            expected_status,
+            request_method=ApiMethod.GET,
+            headers=build_auth_header["header"],
+            json=json,
+        )
+        if expected_status == HTTPStatus.OK:
+            assert expected == response.json
+        elif expected_status == HTTPStatus.CONFLICT:
+            assert {"message": "Dataset indexing was disabled"} == response.json
+        else:
+            assert {
+                "message": "Unauthenticated client is not authorized to READ a resource owned by drb with private access"
+            } == response.json
+
+    @pytest.mark.parametrize("value", (None, "not-integer"))
+    def test_bad_get(self, client, query_api, get_token_func, value):
+        """Check update with no Elasticsearch documents"""
```

Comment:
Hmm; the `if int(count) == 0` actually came along from somewhere else, and I probably should have dropped it during development. If we have an index map, there has to be at least one indexed document: and if we don't have an index map, the `GET` would have failed out at `get_index` in `assemble`. This is actually an internal error. Which of course doesn't mean it can't be mocked ...